exchanges/orderbook/orderbook.go

```diff
@@ -89,89 +88,49 @@ func (s *Service) DeployDepth(exchange string, p currency.Pair, a asset.Item) (*
 	if !a.IsValid() {
 		return nil, errAssetTypeNotSet
 	}
-	mapKey := key.PairAsset{
-		Base:  p.Base.Item,
-		Quote: p.Quote.Item,
-		Asset: a,
-	}
 
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	m1, ok := s.books[strings.ToLower(exchange)]
+	s.m.RLock()
+	book, ok := s.orderbooks[key.ExchangePairAsset{Exchange: exchange, Base: p.Base.Item, Quote: p.Quote.Item, Asset: a}]
+	s.m.RUnlock()
+	var err error
 	if !ok {
-		id, err := s.Mux.GetID()
-		if err != nil {
-			return nil, err
-		}
-		m1 = Exchange{
-			m:  make(map[key.PairAsset]*Depth),
-			ID: id,
-		}
-		s.books[strings.ToLower(exchange)] = m1
+		book, err = s.trackNewBook(&Base{Exchange: exchange, Pair: p, Asset: a})
 	}
-	book, ok := m1.m[mapKey]
-	if ok {
-		// Maybe in future we should return an error here and be more strict.
-		return book, nil
-	}
-	book = NewDepth(m1.ID)
-	book.exchange = exchange
-	book.pair = p
-	book.asset = a
-	m1.m[mapKey] = book
-	return book, nil
+	return book.Depth, err
 }
 
 // GetDepth returns the actual depth struct for potential subsystems and strategies to interact with
-func (s *Service) GetDepth(exchange string, p currency.Pair, a asset.Item) (*Depth, error) {
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	m1, ok := s.books[strings.ToLower(exchange)]
+func (s *store) GetDepth(exchange string, p currency.Pair, a asset.Item) (*Depth, error) {
+	s.m.RLock()
+	book, ok := s.orderbooks[key.ExchangePairAsset{Exchange: exchange, Base: p.Base.Item, Quote: p.Quote.Item, Asset: a}]
+	s.m.RUnlock()
 	if !ok {
-		return nil, fmt.Errorf("%w for %s exchange", ErrOrderbookNotFound, exchange)
+		return nil, fmt.Errorf("%w for %s %s %s", ErrOrderbookNotFound, exchange, p, a)
 	}
-
-	book, ok := m1.m[key.PairAsset{
-		Base:  p.Base.Item,
-		Quote: p.Quote.Item,
-		Asset: a,
-	}]
-	if !ok {
-		return nil, fmt.Errorf("%w associated with base currency %s", ErrOrderbookNotFound, p.Quote)
-	}
-	return book, nil
+	return book.Depth, nil
 }
 
 // Retrieve gets orderbook depth data from the stored tranches and returns the
 // base equivalent copy
-func (s *Service) Retrieve(exchange string, p currency.Pair, a asset.Item) (*Base, error) {
+func (s *store) Retrieve(exchange string, p currency.Pair, a asset.Item) (*Base, error) {
```

Comment:
All `(s *store)` funcs can be unexported with no issue. Your change to an internal store type sets that the order book store is a singleton only. Singleton interactions are handled by `orderbook.Butts()` style functions
yay 🌈

Reply:
unnecessary pedantry

Reply:
no u